src/core/db/backend/pg.py

```diff
@@ -764,3 +768,158 @@ def import_file_w_dbtruck(self, table_name, file_path):
+        for param in params[2:]:
+            self._check_for_injections(param)
+
+        if not self.check_access_permissions(grantor, repo_base):
+            raise Exception('%s does not have permission to define security '
+                            'policies on %s.%s.' % (grantor, repo, table))
+
+        # Raise an exception if the security policy already exists in the table
+        security_policy = self.find_security_policy(
+            table, repo, repo_base, policy=policy, policy_type=policy_type, grantee=grantee,
+            grantor=grantor)
+        if security_policy != []:
+            raise Exception('Security policy already exists in table.')
+
+        query = ('INSERT INTO dh_public.policy (policy, policy_type, grantee, grantor,'
+                 'table_name, repo, repo_base) values '
+                 '(\'%s\', \'%s\', \'%s\', \'%s\', \'%s\', \'%s\', \'%s\')')
+        params = tuple(map(lambda x: AsIs(x), params))
+
+        res = self.execute_sql(query, params, row_level_security=False)
+        return res['status']
+
+    def list_security_policies(self, table, repo, repo_base):
+        '''
+        Return a list of all the security policies defined for the
+        specified table.
+        '''
+        params = [table, repo, repo_base]
+        params = [self.escape_quotes(param) for param in params]
+        for param in params:
+            self._check_for_injections(param)
+
+        query = ('SELECT policy_id, policy, policy_type, grantee, grantor '
+                 'FROM dh_public.policy '
+                 'WHERE table_name = \'%s\' '
+                 'AND repo = \'%s\' AND repo_base = \'%s\'')
+        params = tuple(map(lambda x: AsIs(x), params))
+
+        res = self.execute_sql(query, params, row_level_security=False)
+        return res['tuples']
+
+    def find_security_policy(self, table_name, repo, repo_base,
+                             policy_id=None, policy=None, policy_type=None,
+                             grantee=None, grantor=None):
+        '''
+        Returns a list of all security polices that match the inputs specied
+        by the user.
+        '''
+        params = locals()
+        del params["self"]
+        query = ('SELECT policy_id, policy, policy_type, grantee, grantor '
+                 'FROM dh_public.policy WHERE ')
+
+        first_conditional_added = False
+        for key, value in params.items():
+            if value is None:
+                del params[key]
+                continue
+            value = self.escape_quotes(value)
+            if key != 'policy':
+                self._check_for_injections(str(value))
+
+            if not first_conditional_added:
+                query += '%s = \'%s\'' % (key, value)
+                first_conditional_added = True
+            else:
+                query += ' AND %s = \'%s\'' % (key, value)
+
+        res = self.execute_sql(query, row_level_security=False)
```

Comment:
There's an anti-pattern here that we really shouldn't support. It's far safer to pass params to the execute_sql method. Even if those parameters make use of `AsIs`, it makes it explicit what is and isn't being escaped.

http://initd.org/psycopg/docs/usage.html#the-problem-with-the-query-parameters